packages/core/src/App/Components/Layout/Header/wallets/account-info-wallets.tsx

```diff
@@ -0,0 +1,169 @@
+                                />
+                            )}
+                        </div>
+                    )}
+                    {(typeof active_account?.balance !== 'undefined' || !active_account?.currency) && (
+                        <div className='acc-info__wallets-account-type-and-balance'>
+                            <p
+                                data-testid='dt_balance'
+                                className={classNames('acc-info__balance', {
+                                    'acc-info__balance--no-currency':
+                                        !active_account?.currency && !active_account?.is_virtual,
+                                    'acc-info__wallets-is-virtual-balance':
+                                        active_account?.currency && active_account?.is_virtual,
+                                })}
+                            >
+                                {!active_account?.currency ? (
+                                    <Localize i18n_default_text='No currency assigned' />
+                                ) : (
+                                    `${formatMoney(active_account?.currency, active_account?.balance, true)} ${
+                                        active_account?.currency_config?.display_code
+                                    }`
+                                )}
+                            </p>
+                        </div>
+                    )}
+                    {active_account?.is_disabled ? (
+                        <Icon data_testid='dt_lock_icon' icon='IcLock' />
+                    ) : (
+                        <Icon
+                            data_testid='dt_select_arrow'
+                            icon='IcChevronDownBold'
+                            className='acc-info__select-arrow'
+                        />
+                    )}
+                </div>
+            </AccountInfoWrapper>
+            {is_mobile ? (
+                <AccountSwitcherMobile
+                    is_visible={is_dialog_on}
+                    disableApp={disableApp}
+                    enableApp={enableApp}
+                    toggle={toggleDialog}
+                />
+            ) : (
+                <CSSTransition
+                    in={is_dialog_on}
+                    timeout={200}
+                    classNames={{
+                        enter: 'acc-switcher__wrapper--enter',
+                        enterDone: 'acc-switcher__wrapper--enter-done',
+                        exit: 'acc-switcher__wrapper--exit',
+                    }}
+                    unmountOnExit
+                >
+                    <div className='acc-switcher__wrapper'>
+                        <AccountSwitcherWallet is_visible={is_dialog_on} toggle={toggleDialog} />
+                    </div>
+                </CSSTransition>
+            )}
+        </div>
+    );
+});
+
+export default AccountInfoWallets;
```

Comment:
can we breakdown this component into sub components so we can make more readable